lib/sps_king/message.rb

```diff
@@ -21,22 +22,30 @@ class Message
     def initialize(account_options = {})
       @grouped_transactions = {}
       @account = account_class.new(account_options)
+      @schema_name = get_schema
     end
 
     def add_transaction(options)
+      options[:creditor_address]&.schema_version = account.schema_version(schema_type)
```

Comment:
How is setting the schema on the address at this point going to help when it is only going to be defined on `.to_xml` call?